src/main/java/edu/kis/vh/stacks/StackFIFO.java

```diff
@@ -1,21 +1,44 @@
 package edu.kis.vh.stacks;
 
-public class StackFIFO extends stack {
+import edu.kis.vh.stacks.stackRealization.StackArray;
+import edu.kis.vh.stacks.stackRealization.StackList;
 
-	public stack temp = new stack();
+public class StackFIFO extends Stack {
 	
+// temp zostal zmieniony na StackList poniewaz przy duzej liczbe
+// elementow domyslna wielkosc tablicy (czyli 12) uniemozliwilaby realizacje.
+// Przy kazdej operacji pop konieczne byloby alokowanie nowej tablicy
+// na potrzeby przpisania i odworcenia kolejnosci
+	
+	public StackFIFO() {
+		super();
+	}
+
+	public StackFIFO(int nrOfElements) {
+		super(nrOfElements);
+	}
+
+	public StackFIFO(StackArray stackArrray) {
+		super(stackArrray);
+	}
```

Comment:
Konstruktor powinien przyjmować IStackImplemntation - przecież Array i list oba go implemntują - kłuci się to potem z diagramem który Pan narysował

Reply:
Proszę o przejrzenie testów.